File: Prosit-6/Correction/Solution Jpp CSharp/SQLAccess.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PersonnesEtAdresses
{
    public class SQLAccess : ISQLAccess
    {
        private readonly string _connectionString;
        public SQLAccess(string connectionString)
        {
            _connectionString = connectionString;
        }

        public DataSet GetRows(string sql, string dataTableName)
        {
            DataSet result = new DataSet();
            using (var sqlConnection = new SqlConnection(_connectionString))
            using (var sqlCommand = new SqlCommand(sql, sqlConnection))
            using (var sqlDataAdapter = new SqlDataAdapter(sqlCommand))
            {
                sqlDataAdapter.Fill(result, dataTableName);
            }
            return result;
        }

        public int ExecuteScalar(string sql)
        {
            using (var sqlConnection = new SqlConnection(_connectionString))
            using (var sqlCommand = new SqlCommand(sql, sqlConnection))
            {
                sqlConnection.Open();
                var result = sqlCommand.ExecuteScalar();
                return (int)result;
            }
        }

        public void ExecuteNonQuery(string sql)
        {
            using (var sqlConnection = new SqlConnection(_connectionString))
            using (var sqlCommand = new SqlCommand(sql, sqlConnection))
            {
                sqlConnection.Open();
                sqlCommand.ExecuteNonQuery();
            }
        }
    }
}

File: Prosit-6/Correction/Solution Jpp Cpp/SQLAccess.cpp
#include "pch.h"
#include "SQLAccess.h"

using namespace System;
using namespace System::Data;
using namespace System::Data::Sql;
using namespace System::Data::SqlClient;

namespace PersonnesEtAdresses
{

	SQLAccess::SQLAccess(String^ connectionString) : _connectionString(connectionString)
	{
	}

	DataSet^ SQLAccess::GetRows(String^ sql, String^ dataTableName)
	{
		DataSet^ result = gcnew DataSet();
		SqlConnection^ sqlConnection = gcnew SqlConnection(_connectionString);
		SqlCommand^ sqlCommand = gcnew SqlCommand(sql, sqlConnection);
		SqlDataAdapter^ sqlDataAdapter = gcnew SqlDataAdapter(sqlCommand);
		sqlDataAdapter->Fill(result, dataTableName);
		delete sqlConnection;
		return result;
	}

	int SQLAccess::ExecuteScalar(String^ sql)
	{
		SqlConnection^ sqlConnection = gcnew SqlConnection(_connectionString);
		SqlCommand^ sqlCommand = gcnew SqlCommand(sql, sqlConnection);
		sqlConnection->Open();
		Object^ result = sqlCommand->ExecuteScalar();
		delete sqlConnection;
		return safe_cast<int>(result);
	}

	void SQLAccess::ExecuteNonQuery(String^ sql)
	{
		SqlConnection^ sqlConnection = gcnew SqlConnection(_connectionString);
		SqlCommand^ sqlCommand = gcnew SqlCommand(sql, sqlConnection);
		sqlConnection->Open();
		sqlCommand->ExecuteNonQuery();
		delete sqlConnection;
	}
}
